src/contracts/pods/DelayedWithdrawalRouter.sol

```diff
@@ -107,14 +107,39 @@ contract DelayedWithdrawalRouter is Initializable, OwnableUpgradeable, Reentranc
     }
 
+    /// @notice Getter function to get all delayedWithdrawals of the `user`
+    function getUserDelayedWithdrawals(address user) external view returns (DelayedWithdrawal[] memory) {
+        uint256 delayedWithdrawalsCompleted = _userWithdrawals[user].delayedWithdrawalsCompleted;
+        uint256 totalDelayedWithdrawals = _userWithdrawals[user].delayedWithdrawals.length;
+        uint256 userDelayedWithdrawalsLength = totalDelayedWithdrawals - delayedWithdrawalsCompleted;
+        DelayedWithdrawal[] memory userDelayedWithdrawals = new DelayedWithdrawal[](userDelayedWithdrawalsLength);
+        for (uint256 i = 0; i < userDelayedWithdrawalsLength; i++) {
+            userDelayedWithdrawals[i] = _userWithdrawals[user].delayedWithdrawals[delayedWithdrawalsCompleted + i];
+        }
+        return userDelayedWithdrawals;
+    }
+
     /// @notice Getter function to get all delayedWithdrawals that are currently claimable by the `user`
-    function claimableUserDelayedWithdrawals(address user) external view returns (DelayedWithdrawal[] memory) {
+    function getClaimableUserDelayedWithdrawals(address user) external view returns (DelayedWithdrawal[] memory) {
         uint256 delayedWithdrawalsCompleted = _userWithdrawals[user].delayedWithdrawalsCompleted;
-        uint256 delayedWithdrawalsLength = _userWithdrawals[user].delayedWithdrawals.length;
-        uint256 claimableDelayedWithdrawalsLength = delayedWithdrawalsLength - delayedWithdrawalsCompleted;
-        DelayedWithdrawal[] memory claimableDelayedWithdrawals = new DelayedWithdrawal[](claimableDelayedWithdrawalsLength);
-        for (uint256 i = 0; i < claimableDelayedWithdrawalsLength; i++) {
-            claimableDelayedWithdrawals[i] = _userWithdrawals[user].delayedWithdrawals[delayedWithdrawalsCompleted + i];
+        uint256 totalDelayedWithdrawals = _userWithdrawals[user].delayedWithdrawals.length;
+        uint256 userDelayedWithdrawalsLength = totalDelayedWithdrawals - delayedWithdrawalsCompleted;
+        uint256[] memory claimableDelayedWithdrawalIndices = new uint256[](userDelayedWithdrawalsLength);
+
+        uint256 count = 0;
+        for (uint256 i = 0; i < userDelayedWithdrawalsLength; i++) {
+            DelayedWithdrawal memory delayedWithdrawal = _userWithdrawals[user].delayedWithdrawals[delayedWithdrawalsCompleted + i];
+            if (block.number > delayedWithdrawal.blockCreated + withdrawalDelayBlocks) {
+                claimableDelayedWithdrawalIndices[count] = i;
+                count++;
+            }
+        }
+        
+        DelayedWithdrawal[] memory claimableDelayedWithdrawals = new DelayedWithdrawal[](count);
+        for (uint256 i = 0; i < count; i++) {
+            uint256 index = claimableDelayedWithdrawalIndices[i];
+            DelayedWithdrawal memory delayedWithdrawal = _userWithdrawals[user].delayedWithdrawals[delayedWithdrawalsCompleted + index];
+            claimableDelayedWithdrawals[i] = delayedWithdrawal;
```

Comment:
OK so I think maybe there was some confusion here, now looking at this updated code. all withdrawals are strictly in order. we shouldn't need to store an array of `claimableDelayedWithdrawalIndices` -- this is always just going to be a list starting with the index of the first non-claimed withdrawal, and containing all subsequent indices in order before the first unclaimable one